Specifications/Services/ReverseCall/for_PingedConnection/when_pings_are_requested_every_12_seconds/and_next_message_is_recieved_after_35_seconds.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Dolittle.Runtime.Services.ReverseCalls.for_PingedConnection.given;
+using Machine.Specifications;
+
+namespace Dolittle.Runtime.Services.ReverseCalls.for_PingedConnection.when_pings_are_requested_every_12_seconds
+{
+    public class and_next_message_is_recieved_after_35_seconds : given.all_dependencies
+    {
+        Establish context = () =>
+        {
+            scenario = Scenario.New(_ => {
+                _.Receive.Message(first_message_with_12_second_pings).AtTime(20);
+                _.Receive.Message(new()).AtTime(55);
+            });
+        };
+
+        Because of = () => scenario.Simulate(
+            request_id,
+            server_call_context,
+            message_converter.Object,
+            metrics,
+            logger_factory);
+
+        It should_have_written_a_pings = () => scenario.WrittenMessageTimes.ShouldContainOnly(20, 32, 44);
+        It should_have_set_the_initial_refresh_time = () => scenario.RefreshedTokenTimes.ShouldContainOnly(20, 55);
```

Comment:
A minor thing, but instead of literal numbers we could use variables/consts for the values here. That way we don't have to worry about typoing or miscalculating something and it becomes more apparent what is being calculated here.

For example we could do something like this:

```suggestion
            scenario = Scenario.New(_ => {
                _.Receive.Message(first_message_with_12_second_pings).AtTime(start_time);
                _.Receive.Message(new()).AtTime(start_time + interval);
            });
        };

        Because of = () => scenario.Simulate(
            request_id,
            server_call_context,
            message_converter.Object,
            metrics,
            logger_factory);

        // not sure about the syntax here, but you get the point
        It should_have_written_a_pings = () => scenario.WrittenMessageTimes.ShouldContainOnly(Enumerable.Range(0, 3).Select(i => start_time + ping_interval * i);
        It should_have_set_the_initial_refresh_time = () => scenario.RefreshedTokenTimes.ShouldContainOnly(start_time, start_time + interval);
```

I think that example is an overkill for this situation (especially the ugly `Enumerable.Range.Select`), but maybe some of the values could be moved to their respective `given`'s or static fields, like the 12s and 35s. It also think it would help readability (apparently 1st grade math is too crazy for me haha 😆)

Reply:
I agree that it requires a bit of thinking to make sure the numbers here make sense - but I don't think putting them in variables makes it that much easier to read. As essentially every number would either be a variable or expression. IMO this is just some of those specs where you have to think about what is going on, and suggest that we leave it as is. If you still feel like it would be nicer with variables - feel free to commit commit a suggestion, you might be seeing something I don't :)

Reply:
I think it's fine as it is for a spec 👍, I don't think we need to work more on it now. If C# would have simpler range/array syntax like Python or JS I think it would make more sense too.